third_party/cares/cares.BUILD

```diff
@@ -138,10 +144,22 @@ cc_library(
     copts = [
         "-D_GNU_SOURCE",
         "-D_HAS_EXCEPTIONS=0",
-        "-DNOMINMAX",
         "-DHAVE_CONFIG_H",
-    ],
+    ] + select({
+        ":windows": [
+            "-DNOMINMAX",
+            "-D_CRT_SECURE_NO_DEPRECATE",
```

Comment:
same with these two macros: `_CRT_SECURE_NO_DEPRECATE` and `_CRT_NONSTDC_NO_DEPRECATE` - can you please explain?

Reply:
`NOMINMAX` is to tell `windows.h` not to define `min` and `max` macros. These two macros often cause obscure compile error if you something else with the same name.

`_CRT_SECURE_NO_DEPRECATE` and `_CRT_NONSTDC_NO_DEPRECATE` are needed because MSVC is trying to deprecate perfectly normal C standard functions such as `fopen` (and ask you to use `fopen_s` instead). Even gRPC use these flags: https://github.com/grpc/grpc/blob/master/CMakeLists.txt#L99.

tl;dr: We need these three flags to avoid useless warnings.

Reply:
Yeah, these are very standard for Windows building.